rasa/nlu/components.py

```diff
@@ -208,18 +209,36 @@ def validate_pipeline(pipeline: List["Component"]) -> None:
     validate_required_components(pipeline)
 
 
-def any_components_in_pipeline(components: Iterable[Text], pipeline: List["Component"]):
+def any_components_in_pipeline(
+    components: Iterable[Text], pipeline: List["Component"]
+) -> bool:
     """Check if any of the provided components are listed in the pipeline.
 
     Args:
-        components: A list of :class:`rasa.nlu.components.Component`s to check.
+        components: A list of str of component class names to check
         pipeline: A list of :class:`rasa.nlu.components.Component`s.
 
     Returns:
         `True` if any of the `components` are in the `pipeline`, else `False`.
 
     """
-    return any(any(component.name == c for component in pipeline) for c in components)
+    return len(find_components_in_pipeline(components, pipeline)) > 0
+
+
+def find_components_in_pipeline(
+    components: Iterable[Text], pipeline: List["Component"]
+) -> List[str]:
+    """Finds those of the given components that are present in the pipeline.
+
+    Args:
+        components: A list of str of component class names to check.
+        pipeline: A list of :class:`rasa.nlu.components.Component`s.
+
+    Returns:
+        A list of str of component class names that are present in the pipeline.
+    """
+    pipeline_component_names = set([c.name for c in pipeline])
+    return list(pipeline_component_names.intersection(components))
```

Comment:
do we actually need the string cast?

Reply:
You meant set cast right? I simplified that method a bit

Reply:
Sorry, meant the `list(...)` cast 🙈